tracer/src/Datadog.Trace/AppSec/Concurrency/ReaderWriterLock.Core.cs

```diff
@@ -25,11 +26,11 @@ internal bool EnterReadLock()
         return true;
     }
 
-    internal bool EnterWriteLock()
+    internal bool EnterWriteLock(int timeout = TimeoutInMs)
```

Comment:
nit: should we make the same change to `EnterReadLock` for consistency?

Reply:
I'm not sure... we don't need to have different read timeouts for now.... 🤔  we needed this for the dispose scenario, to make sure it ends up disposing the handle but don't want to give incentive to specify different read timeouts to run the waf otherwise 🤔 ..